src/sync/sync_manager.cpp

```diff
@@ -387,6 +393,12 @@ std::shared_ptr<SyncSession> SyncManager::get_existing_session_locked(const std:
     return it->second;
 }
 
+std::shared_ptr<SyncSession> SyncManager::get_existing_session(const std::string& path) const
+{
+    std::lock_guard<std::mutex> lock(m_session_mutex);
+    return get_existing_session_locked(path);
```

Comment:
This needs to return an external reference (`SyncSession::external_reference`). I suspect it also needs to attempt to revive the session by calling `SyncUser::register_session` so it doesn't return a session that is in the process of dying.